Mathlib/Tactic/Linter/AdmitLinter.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa, Adomas Baliuka
+-/
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+# The "admit" linter
+
+The "admit" linter flags usages of the `admit` tactic.
+
+The tactics `admit` and `sorry` are synonyms.
+The use of `sorry` is much more common and should be preferred.
+
+This linter is an incentive to discourage uses of `admit`, without being a ban.
+-/
+
+open Lean Elab
+
+namespace Mathlib.Linter.admit
+
+/-- The admit linter emits a warning on usages of `admit`. -/
+register_option linter.admit : Bool := {
+  defValue := true
+  descr := "enable the admit linter"
+}
+
+syntax (name := admit) "admit " : tactic
+/-- `admit` is a shorthand for `exact sorry`. -/
+macro "admit" : tactic => `(tactic| exact @sorryAx _ false)
+
+/-- `getadmit t` returns all usages of the `admit` tactic in the input syntax `t`. -/
+partial
+def getadmit : Syntax → Array Syntax
+  | stx@(.node _ kind args) =>
+    let rargs := (args.map getadmit).flatten
+    if kind == ``admit then rargs.push stx else rargs
```

Comment:
```suggestion
    if kind == ``Lean.Parser.Tactic.tacticAdmit then rargs.push stx else rargs
```
Then you can delete your `admit` definition.

Or, even better, use a syntax quotation like
```lean
partial
def getadmit (stx : Syntax) : Array Syntax :=
  if let `(tactic| admit) := stx then
    #[stx]
  else if let .node _ _ args := stx then
    (args.map getadmit).flatten
  else
    #[]
```
Even shorter:
```lean
partial
def getadmit (stx : Syntax) : Array Syntax :=
  if let `(tactic| admit) := stx then
    #[stx]
  else
    stx.foldArgs (fun arg r => r ++ getadmit arg) #[]
```

Reply:
Used the last suggestion.